src/main/kotlin/uk/gov/justice/digital/hmpps/visitscheduler/controller/MigrateController.kt

```diff
@@ -0,0 +1,69 @@
+package uk.gov.justice.digital.hmpps.visitscheduler.controller
+
+import io.swagger.v3.oas.annotations.Operation
+import io.swagger.v3.oas.annotations.media.Content
+import io.swagger.v3.oas.annotations.media.Schema
+import io.swagger.v3.oas.annotations.responses.ApiResponse
+import org.springframework.http.HttpStatus
+import org.springframework.http.MediaType
+import org.springframework.security.access.prepost.PreAuthorize
+import org.springframework.validation.annotation.Validated
+import org.springframework.web.bind.annotation.PostMapping
+import org.springframework.web.bind.annotation.RequestBody
+import org.springframework.web.bind.annotation.RequestMapping
+import org.springframework.web.bind.annotation.ResponseStatus
+import org.springframework.web.bind.annotation.RestController
+import uk.gov.justice.digital.hmpps.visitscheduler.config.ErrorResponse
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.MigrateVisitRequestDto
+import uk.gov.justice.digital.hmpps.visitscheduler.dto.VisitDto
+import uk.gov.justice.digital.hmpps.visitscheduler.service.MigrateVisitService
+import javax.validation.Valid
+
+@RestController
+@Validated
+@RequestMapping(name = "Visit Resource", path = ["/migrate-visits"], produces = [MediaType.APPLICATION_JSON_VALUE])
+class MigrateController(
+  private val migrateVisitService: MigrateVisitService
+) {
+
+  @PreAuthorize("hasRole('VISIT_SCHEDULER')")
+  @PostMapping
+  @ResponseStatus(HttpStatus.CREATED)
+  @Operation(
+    summary = "Migrate a visit",
+    requestBody = io.swagger.v3.oas.annotations.parameters.RequestBody(
+      content = [
+        Content(
+          mediaType = "application/json",
+          schema = Schema(implementation = MigrateVisitRequestDto::class)
+        )
+      ]
+    ),
+    responses = [
+      ApiResponse(
+        responseCode = "201",
+        description = "Visit migrated"
+      ),
+      ApiResponse(
+        responseCode = "400",
+        description = "Incorrect request to migrate a visit",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "401",
+        description = "Unauthorized to access this endpoint",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      ),
+      ApiResponse(
+        responseCode = "403",
+        description = "Incorrect permissions to migrate a visit",
+        content = [Content(mediaType = "application/json", schema = Schema(implementation = ErrorResponse::class))]
+      )
+    ]
+  )
+  fun migrateVisit(
+    @RequestBody @Valid migrateVisitRequest: MigrateVisitRequestDto
+  ): VisitDto {
```

Comment:
Should this be a LegacyVisitDto or such?  return the complete dto now we don't have to hide elements from VSiP

Reply:
Not even sure if we need to return the DTO,... may be just the reference...